### PR TITLE
Az.Tools.Installer v2

### PR DESCRIPTION
**Pull request by msJinLei**
## Description
1. Download the nupkgs from remote repo to local and install module from
local
2. Use thread job for PowerShell core and check job status and error
3. Remove MininumVersion and MaximumVersion of the cmdlets and simplify
the logic of uninstallation
4. Support the installation of a url or a path
5. Display progress bar

The replacement for https://github.com/Azure/azure-powershell/pull/16277

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by dingmeng-xue** [comment]
The file tools/Az.Tools.Installer/test/Az.Tools.Installer-TestResults.xml should be removed. Please add .gitignore to exclude it.

Another recommendation is that you need to add description to each example which can help user to understand the purpose of script.

File: tools/Az.Tools.Installer/Az.Tools.Installer.psd1 [hunk 1]
```diff
@@ -109,8 +109,12 @@
             # IconUri = ''
     
             # ReleaseNotes of this module
-            ReleaseNotes = '* Fix bugs for Install-AzModule
-            * Fix syntax errors when run in Windows Powershell'
+            ReleaseNotes = '* Supported for installing from a path of nuget package
```
Comment (dingmeng-xue) [inline tools/Az.Tools.Installer/Az.Tools.Installer.psd1:112]:
```suggestion
            ReleaseNotes = '* Supported installing from a path of nuget package
```

File: tools/Az.Tools.Installer/Az.Tools.Installer.psd1 [hunk 2]
```diff
@@ -109,8 +109,12 @@
...
+            * Supported installing Az sub modules using minimum required Az.Accounts version
+            * Supported uninstalling by module name filter and whether in preview
+            * Provided progress bar to show the cmdlet execution status
+            * Unsupported uninstalling by Az version
```
Comment (dingmeng-xue) [inline tools/Az.Tools.Installer/Az.Tools.Installer.psd1:116]: This line is not required.

**Comment by msJinLei** [comment]
> The file tools/Az.Tools.Installer/test/Az.Tools.Installer-TestResults.xml should be removed. Please add .gitignore to exclude it.
> 
> Another recommendation is that you need to add description to each example which can help user to understand the purpose of script.

Done

**Review by dingmeng-xue — APPROVED** [review]

**Pull request merged**